agents/agent.py

```diff
@@ -156,14 +140,40 @@ def from_config_file(cls, filename=None, log_path=None):
         Returns:
             ServerAgent: Child instance of ServerAgent.
         """
-        agent = cls(log_path=log_path)
+        agent = cls()
 
         agent._parse_config_file(filename)
+        agent.setup_logging(log_path)
 
         return agent
 
+    def setup_logging(self, log_path=None):
+        """
+        Setup agent's logger based on its server_name.
+
+        Parameters:
+            log_path (PathLike, optional): Path to where logs will be
+                stored.
+        """
+        log_path = (
+            log_path or pkg_resources.resource_filename(
+                self.__class__.__module__, 'logs/%s.log' % self.server_name
+            )
+        )
+
+        logging.config.fileConfig(
```

Comment:
`fileConfig()` completely overrides existing logging configuration. Consider using `dictConfig()` if you need more granular or additive control, especially if you anticipate future logging customization.